source/views/menus/menu-bonapp.js

```diff
@@ -261,11 +261,6 @@ export class BonAppHostedMenu extends React.PureComponent<Props, State> {
 			return <NoticeView text={msg} />
 		}
 
-		if (Object.keys(this.state.cafeMenu.items).length === 0) {
```

Comment:
Right, so I don't think we can outright remove this. I added it originally back in #3047 because we have to handle all ranges of cafes provided by bonapp. Sometimes there are no dayparts and no items, yet the cafeteria is open.

I'm testing this with cafe id 2 and the error that this block of code originally helped catch is back without it. So while it may cause another issue where it's at now, this still needs to be taken into account.

Reply:
Okay. I'm noticing that we are unable to parse cafe ids 0 and 3000. The cafes returns `{"cafe":[]}`. We should add a check for no cafe data...

Reply:
Good point! The latest commit should resolve this..

Reply:
I don’t know how many edge cases this spans to

Reply:
I guess this actually covers it 👍